api/specs.proto

```diff
@@ -270,4 +275,19 @@ message ClusterSpec {
 
 	// TaskDefaults specifies the default values to use for task creation.
 	TaskDefaults task_defaults = 7 [(gogoproto.nullable) = false];
+
+	// SecretConfig specifies cluster-level secret configuration settings, such as
+	// maximum number of secrets.
+	SecretsConfig secrets = 8 [(gogoproto.nullable) = false];
+}
+
+// SecretSpec specifies a user-provided secret.
+message SecretSpec {
```

Comment:
Should the `SecretSpec` be here? Maybe this PR should be rebased out of #1511 so it doesn't include these portions?